winconf.pri

```diff
@@ -41,6 +38,7 @@ win32-g++* {
 }
 else:win32-msvc* {
     CONFIG -= embed_manifest_exe
+    QMAKE_CXXFLAGS += /std:c++17
```

Comment:
How avout other compilers?

Reply:
We have 3 ways for building:
1. cmake
  This is for every compiler.
2. autotools + qmake
  This is for linux-like compilers (gcc and mingw).
3. qmake
  AFAIK, only MSVC uses this way.

IMO all paths are already handled correctly.

Reply:
>autotools + qmake
This is for linux-like compilers (gcc and mingw).

Personally I never used MinGW GCC with autotools. It's just a yet another compiler like MSVC.
Even if you use it with autotools, then I still don't see any reason why we can't have this setting here (in QMake project), because c++17 is the only option that can't be set/changed during configuration, isn't it?

Reply:
>then I still don't see any reason why we can't have this setting here (in QMake project), 

The complex "c++ mode detection" code in `autotools` exists for a reason (it is hard to explain in a few words...), I don't want to strip it down. And since autotools is already handling it, it is best not to force it in qmake.

Or do you suggest I add `QMAKE_CXXFLAGS += -std=c++17 to the `win32-g++*` case, I think it is not unreasonable.

Reply:
>Or do you suggest I add `QMAKE_CXXFLAGS += -std=c++17` to the `win32-g++*` case? I think it is not unreasonable.

Yes.

Reply:
Or maybe it should be handled by conf.pri instead? In all the cases?

Reply:
> > Or do you suggest I add `QMAKE_CXXFLAGS += -std=c++17` to the `win32-g++*` case? I think it is not unreasonable.
> 
> Yes.

Done, in some other way.

>Or maybe it should be handled by conf.pri instead? 

I think this is limited to Windows platform, so no.

Reply:
>Done, in some other way.

At least Qt-5.9 (and I think some later versions) doesn't know about c++17 value for CONFIG.

Reply:
>>Or maybe it should be handled by conf.pri instead?
>>
>I think this is limited to Windows platform, so no.

Don't see any problems... You just need to add it in `conf.pri.windows`.

Reply:
>You just need to add it in conf.pri.windows.

But that file is a template for customization, strict requirement (such as C++17) shouldn't be placed into that file, isn't?

Reply:
👍

Reply:
>At least Qt-5.9 (and I think some later versions) doesn't know about c++17 value for CONFIG.

Indeed, reverted to use `QMAKE_CXXFLAGS`.